network/gossip/libp2p/libp2pNode.go

```diff
@@ -232,15 +230,9 @@ func (p *P2PNode) RemovePeer(ctx context.Context, peer NodeAddress) error {
 		return fmt.Errorf("failed to remove peer %s: %w", peer.Name, err)
 	}
 
-	p.Lock()
-	defer p.Unlock()
-	conns := p.libP2PHost.Network().ConnsToPeer(pInfo.ID)
-	// close all connections with the peer
-	for _, c := range conns {
-		err = c.Close()
-		if err != nil {
-			return fmt.Errorf("failed to remove peer %s: %w", peer.Name, err)
-		}
+	err = p.libP2PHost.Network().ClosePeer(pInfo.ID)
```

Comment:
`.ClosePeer` has the same effect of looping through all connections and closing them individually.